### DISCUSSION TITLE
how does netbox get device information

### DISCUSSION THREAD

**Discussion by robina80** [netbox-community/netbox#14880] [General]
hi all.

ive installed netbox and i spose i just add all my inventory to it like switches FW's computers, Wifi etc as i spose it doesnt auto discover my devices and put it in the inventory itself, is this the case, so its a glorified spreadsheet?

thanks,
rob

**Comment by abhi1693** [comment]
NetBox functions as the source of truth for your network infrastructure. Its job is to define and validate the intended state of all network components and resources. NetBox does not interact with network nodes directly; rather, it makes this data available programmatically to purpose-built automation, monitoring, and assurance tools. This separation of duties enables the construction of a robust yet flexible automation system.

**Comment by robina80** [comment]
So it doesn't go on the network and do an auto inventory and grabs devices
and builds a map of the network, you need to enter in manually all your
devices

Is there a tool that can do the above ie auto discovery and add the device
to netbox

**Reply by llamafilm** [reply]
I think you're looking at this backwards.  Netbox is designed to be the source of truth, i.e. the _intended_ state of your network.  You can use other tools to discover and monitor the _actual_ state, and then build some custom integrations using Netbox API to generate reports comparing the two.
One such tool I heard of recently is [LanSweeper](https://www.lansweeper.com) but I haven't tried it myself.
[PRTG](https://www.paessler.com) also has some discovery features.

**Comment by robina80** [comment]
so netbox is your manual inventory or what you think is on your network and then you cross reference it with what is on your actual network using librenms or prtg that actually auto discover whats on your network?

can you get a netbox add on/plugin to get netbox to auto discover and add to the inventory rather than you doing it manually?

**Reply by xkilian** [reply]
We use the output from our RANCID or Ansible config backups/show commands to build our initial bulk input to netbox. For the rest we import our excel ip/prefix plans and then cross reference with nmap or other discovery tool to highlight gaps in documentation.
Once that is done, the it is netbox native management(ui, csv, form based netbox plugins or api)

**Comment by robina80** [comment]
i get it now, so it rely's on other tools like libreNMS, Ansible etc to grab/save the info and save it to netbox?

**Reply by jeremystretch** [reply]
Check out the reference architecture recently added to our README:

![Reference architecture](https://github.com/netbox-community/netbox/assets/13487278/c06f0f27-54ee-4e7f-b21e-e95ac6bdbc78)

NetBox's primary function is to validate infrastructure state defined by operators and make it available programmatically to provisioning and monitoring tools. This approach ensures a very clean separation of duties among tools, and helps combat vendor lock-in.
